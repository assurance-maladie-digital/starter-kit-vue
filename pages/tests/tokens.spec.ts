import Tokens from '../tokens.vue';
import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, it, expect } from 'vitest';

describe('home page should render', () => {
	it('should render', async () => {
		const wrapper = await mountSuspended(Tokens);
		expect(wrapper.html()).toMatchSnapshot();

		const h1Text = wrapper.find('h1').text();
		expect(h1Text).toContain('Tokens');
	});

	it('should return true for dark color', async () => {
		const wrapper = await mountSuspended(Tokens);
		expect(Tokens.methods.isDarkColor('#000000')).toBe(true);
	})

	it('should return true for light color', async () => {
		const wrapper = await mountSuspended(Tokens);
		expect(Tokens.methods.isDarkColor('#ffffff')).toBe(false);
	})
});



