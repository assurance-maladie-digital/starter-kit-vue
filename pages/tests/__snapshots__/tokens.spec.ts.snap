// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home page should render > should render 1`] = `
"<h1 data-v-cf17212c="" class="text-primary">Tokens</h1>
<h2 data-v-cf17212c="" class="py-8">Heading</h2>
<div data-v-cf17212c="" style="background-color: #e0e7f2;" class="text-center">
  <h1 class="my-1" data-v-cf17212c="">h1. Design system heading</h1>
  <h2 class="my-1" data-v-cf17212c="">h2. Design system heading</h2>
  <h3 class="my-1" data-v-cf17212c="">h3. Design system heading</h3>
  <h4 class="my-1" data-v-cf17212c="">h4. Design system heading</h4>
  <h5 class="my-1" data-v-cf17212c="">h5. Design system heading</h5>
  <h6 class="my-1" data-v-cf17212c="">h6. Design system heading</h6>
</div>
<h2 data-v-cf17212c="" class="py-8">Couleurs du theme Vuetify</h2>"
`;

exports[`tokens page should render > should render 1`] = `
"<h1 data-v-cf17212c="" class="text-primary">Tokens</h1>
<h2 data-v-cf17212c="" class="py-8">Heading</h2>
<div data-v-cf17212c="" style="background-color: #e0e7f2;" class="text-center">
  <h1 class="my-1" data-v-cf17212c="">h1. Design system heading</h1>
  <h2 class="my-1" data-v-cf17212c="">h2. Design system heading</h2>
  <h3 class="my-1" data-v-cf17212c="">h3. Design system heading</h3>
  <h4 class="my-1" data-v-cf17212c="">h4. Design system heading</h4>
  <h5 class="my-1" data-v-cf17212c="">h5. Design system heading</h5>
  <h6 class="my-1" data-v-cf17212c="">h6. Design system heading</h6>
</div>
<h2 data-v-cf17212c="" class="py-8">Couleurs du theme Vuetify</h2>"
`;
